components/channel_header/index.js

```diff
@@ -73,6 +74,8 @@ function makeMapStateToProps() {
             isMuted: isCurrentChannelMuted(state),
             isQuickSwitcherOpen: isModalOpen(state, ModalIdentifiers.QUICK_SWITCH),
             hasGuests: stats.guest_count > 0,
+            getDisplayNameByUserId: (userId) => getDisplayNameByUserId(userId, state),
```

Comment:
Same comment here. For this one, we could instead pass in the teammate name display setting (which you can get using `getTeammateNameDisplaySetting` from `mattermost-redux/selectors/entities/preferences`) and have the component call `displayUsername` (from `mattermost-redux/utils/user_utils`)